local/php/envs.go

```diff
@@ -31,17 +31,36 @@ import (
 
 func (p *Server) generateEnv(req *http.Request) map[string]string {
 	scriptName := p.passthru
+
 	https := ""
 	if req.TLS != nil {
 		https = "On"
 	}
 
 	pathInfo := req.URL.Path
+	filenameMatch := false
+
 	if pos := strings.Index(strings.ToLower(pathInfo), ".php"); pos != -1 {
 		file := pathInfo[:pos+4]
 		if _, err := os.Stat(filepath.Join(p.documentRoot, file)); err == nil {
 			scriptName = file
 			pathInfo = pathInfo[pos+4:]
+			filenameMatch = true
+		}
+
+	}
+
+	if filenameMatch == false && pathInfo != "/" && len(pathInfo) > 0 {
+		paths := strings.Split(strings.Trim(pathInfo, "/"), "/")
+		for i, n := 0, len(paths); i < n; i++ {
+			poppedPath := paths[:len(paths)-i]
+			indexDir := filepath.Join(poppedPath...)
+			file := filepath.Join(indexDir, p.passthru)
+			if _, err := os.Stat(filepath.Join(p.documentRoot, file)); err == nil {
+				scriptName = string(os.PathSeparator) + file
+				pathInfo = pathInfo[strings.Index(pathInfo, indexDir)+len(indexDir):]
+				break
+			}
 		}
 	}
```

Comment:
would you mind extracting all this into a private function?
early returns would help to have better readability here

Reply:
Good shout!